modules/misc/nixgl.nix

```
@@ -0,0 +1,103 @@
+{ config, pkgs, lib, ... }:
+
+let
+  cfg = config.workarounds.nixgl;
+
+  # Expects a PackageConfiguration hashset
+  # Adapted from https://nixos.wiki/wiki/Nix_Cookbook
+  wrapWithNixGL = { binary, pkg, ... }:
+    (let
+      bin = if binary == null then pkg.pname else binary;
+      nixGL = import cfg.src { inherit pkgs; };
+      wrapperPkg = lib.attrsets.getAttrFromPath cfg.wrapperPkgPath nixGL;
+      wrapperBinPath = "${wrapperPkg}/bin/${cfg.wrapperBinName}";
+    in pkgs.runCommand "nixgl-${pkg.pname}" {
+      buildInputs = [ pkgs.makeWrapper ];
+    } ''
+      mkdir $out
+      # Link every top-level folder from pkgs.hello to our new target
+      ln -s ${pkg}/* $out
+      # Except the bin folder
+      rm $out/bin
+      mkdir $out/bin
+      # We create the bin folder ourselves and link every binary in it
+      ln -s ${pkg}/bin/* $out/bin
+      # Except the target binary
+      rm $out/bin/${bin}
+      # Because we create this ourself, by creating a wrapper
+      makeWrapper ${wrapperBinPath} $out/bin/${bin} \
+        --add-flags ${pkg}/bin/${bin}
+    '');
+  PackageConfiguration = with lib;
+    types.submodule {
+      options = {
+        binary = mkOption {
+          type = types.nullOr types.str;
+          example = "alacritty";
+          default = null;
+          description = ''
+            Name of the binary in bin/ of the package.
+            By default this uses `package.pname`.
+          '';
+        };
+        pkg = mkOption {
+          type = types.package;
+          example = "pkgs.alacritty";
+          description = ''
+            The package to be wrapped.
+            It's expected to have a `pname` attribute.
+          '';
+        };
+      };
```

Can just set the binary option directly in the submodule:
```suggestion
      };
      
      config = {
        binary = mkOptionDefault (config.package.meta.mainProgram or config.package.pname);
      };
```

Can then remove allowing `null` values.

---

I'm not able to get this to work. It says `attribute "package" missing`. I tried `binary = mkOptionDefault (config: (config.package.meta.mainProgram or getName config.package));` hoping the function would be evaluated, but it doesn't.